ui/desktop/src/main.ts

```diff
@@ -447,46 +446,37 @@ const parseArgs = () => {
   return { dirPath };
 };
 
-const getGooseProvider = () => {
-  loadShellEnv(app.isPackaged);
+interface BundledConfig {
+  defaultProvider?: string;
+  defaultModel?: string;
+  predefinedModels?: string;
+  baseUrlShare?: string;
+  version?: string;
+}
+
+const getBundledConfig = (): BundledConfig => {
   //{env-macro-start}//
   //needed when goose is bundled for a specific provider
   //{env-macro-end}//
-  return [
-    process.env.GOOSE_DEFAULT_PROVIDER,
-    process.env.GOOSE_DEFAULT_MODEL,
-    process.env.GOOSE_PREDEFINED_MODELS,
-  ];
-};
-
-const getSharingUrl = () => {
```

Comment:
did you mean to remove all this stuff? doesn't seem related to the change

Reply:
I consolidated the sharingUrl and version into one function, since they were only called in one place next to each other, and all read from the injected config

Reply:
before, if you had e.g. called getSharingUrl or getVersion before getGooseProvider, the injected code wouldn't run until the last one and you'd end up with a tricky bug

Reply:
ah nice catch!